pkg/server/server.go

```diff
@@ -934,6 +936,7 @@ func findCRD(crdName string, crds []*apiextensionsv1.CustomResourceDefinition) (
 			crd = aCRD
 		} else {
 			if !equality.Semantic.DeepEqual(crd.Spec, aCRD.Spec) {
+				klog.V(10).Infof("Found multiple CRDs with the same name %q, but different specs: %v", crdName, cmp.Diff(crd.Spec, aCRD.Spec))
```

Comment:
V(10) means that when we have a rare flake, nobody will see this log, and you won't be able to debug :)

Reply:
`klog.Errorf()` then? I didn't want it to be too spammy, but perhaps in this case it should

Reply:
Info (no v specified) or Error is fine for now. Let's put in a TODO to revisit the logging and/or level after we catch the flake, maybe.